CIP-2551/CIP-2551.md

```diff
@@ -0,0 +1,271 @@
+---
+CIP: ?
+Title: Ed25519 elliptic curve group primitives
+Authors: Morgan Thomas <morgan.thomas@platonic.systems>
+Discussions-To: <morgan.thomas@platonic.systems>
+Status: Draft
+Type: Standards Track
+Created: 2022-07-27
+* License: Apache-2.0
+---
+
+## Preamble
+
+RFC 822 style headers containing metadata about the CIP, including the CIP
+number, a short descriptive title (limited to a maximum of 44 characters),
+the names, and optionally the contact info for each author, etc.
+
+## Simple Summary
+
+This CIP proposes that Plutus be extended with native code
+implementations of certain mathematical operations which are useful for
+cryptography. These mathematical operations are used in the implementation
+of the Ed25519 public/private key signing protocol. Cardano already
+supports a native code implementation of verifying signatures created by
+the Ed25519 signing protocol, but it does not enable Plutus to scripts
+perform the underlying mathematical operations directly. Being able to do
+this would be useful for implementing other cryptographic protocols based
+on the Ed25519 mathematical operations.  Applications include verifying
+zero-knowledge proofs, such as in the context of implementing zk-rollups.
+
+## Abstract
+
+Extend Plutus with primitives for performing calculations over the
+Ed25519 curve and the corresponding finite field.  These operations are
+to include the basic group and field operations, as well as multiscalar
+multiplication.
+
+## Motivation
+
```

Comment:
We also need a motivation of why we use a curve with cofactor 8 (such as the edwards curve) instead of one with cofactor 1 which is usually safer. This is for efficiency, as edwards curves are much more efficient. Nonetheless, I think it is necessary to explain the danger of having a group with cofactor 8. This makes me wonder if we also need built-in functions to check if the points are part of the prime order group or not. What do you think?